content/200-concepts/100-components/01-prisma-schema/06-relations/420-relation-mode.mdx

```diff
@@ -0,0 +1,214 @@
+- `prisma`: this emulates relations in the Prisma client. This is the default (and only available) option when using the MongoDB connector. You should also [enable this option](/guides/database/using-prisma-with-planetscale#how-to-enable-emulation-of-referential-integrity) when using the MySQL connector with a PlanetScale database.
+
+Setting the relation mode is currently a preview feature. To enable this feature, add the `referentialIntegrity` feature flag to the `previewFeatures` list in the [`generator`](/reference/api-reference/prisma-schema-reference#generator) block:
+
+```prisma file=schema.prisma highlight=9;add
+datasource db {
+  provider     = "mysql"
+  url          = env("DATABASE_URL")
+  relationMode = "prisma"
+}
+
+generator js {
+  provider        = "prisma-client-js"
+  previewFeatures = ["referentialIntegrity"]
+}
+```
+
+<Admonition type="warning">
+
+If you switch between relation modes, Prisma will add or remove foreign keys to your database next time you apply changes to your schema with Prisma Migrate or `db push`. See [Switch between relation modes](#switch-between-relation-modes) for more information.
+
+</Admonition>
+
+<!-- TODO: mention preview feature + version info in intro -->
+
+## Handle relations in your relational database with the <inlineCode>foreignKeys</inlineCode> relation mode
+
+The `foreignKeys` relation mode handles relations in your relational database using foreign keys. This is the default option when you use a relational database connector (PostgreSQL, MySQL, SQLite, SQL Server, CockroachDB).
+
+The `foreignKeys` relation mode is not available when you use the MongoDB connector. Some relational databases, [such as PlanetScale](/guides/database/using-prisma-with-planetscale#how-to-emulate-referential-integrity-in-prisma-client), also forbid the use of foreign keys. In these cases, you should instead [emulate relations in Prisma with the `prisma` relation mode](#emulate-relations-in-prisma-with-the-prisma-relation-mode).
+
+### Referential integrity
+
+The `foreignKeys` relation mode enforces referential integrity checks at the database level. This means that when you _create_ or _update_ a record with a relation to another record, the related record needs to exist. If the record does not exist, the database will return an error message.
+
+### Referential actions
+
+When you use the `foreignKeys` relation mode, referential actions are triggered in the database when you _update_ or _delete_ a record with relations to other records. These referential actions maintain referential integrity in the related records by preventing changes that would break referential integrity, cascading changes through to related records or setting the value of fields that reference the updated or deleted record to a `null` or default value.
+
+For more information, see the [referential actions](/concepts/components/prisma-schema/relations/referential-actions) page.
```

Comment:
```suggestion
### Referential integrity

#### Foreign Key Constraints

The `foreignKeys` relation mode enforces foreign key constraints at the database level to maintain referential integrity. This means that when you _create_ or _update_ a record with a relation to another record, the related record needs to exist. If the record does not exist, the database will return an error message.

#### Referential actions

When you use the `foreignKeys` relation mode, referential actions are triggered in the database when you _update_ or _delete_ a record with relations to other records. These referential actions maintain referential integrity in the related records by preventing changes that would break referential integrity, cascading changes through to related records or setting the value of fields that reference the updated or deleted record to a `null` or default value.

For more information, see the [referential actions](/concepts/components/prisma-schema/relations/referential-actions) page.
```

Reply:
Done this (also reworded a bit)